cmake/Settings.cmake

```diff
@@ -0,0 +1,34 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+# Settings intended to be applied to every Swift target in this package.
+# Analogous to project-level build settings in an Xcode project.
+add_compile_definitions(
```

Comment:
Not sure where the right place for it would be, but we should make sure the following Swift compiler conditionals are declared:

```swift
.define("SWT_TARGET_OS_APPLE", .when(platforms: [.macOS, .iOS, .macCatalyst, .watchOS, .tvOS, .visionOS])),

.define("SWT_NO_FILE_IO", .when(platforms: [.wasi])),
.define("SWT_NO_EXIT_TESTS", .when(platforms: [.iOS, .watchOS, .tvOS, .visionOS, .wasi])),
```

Otherwise the code won't build correctly on Apple platforms or WASI.

Reply:
This would be the right place.  I'll just need to use the correct `if()` guards.